src/nvim/eval.c

```diff
@@ -13047,6 +13051,55 @@ static void f_msgpackparse(typval_T *argvars, typval_T *rettv)
   return;
 }
 
+/// "msgpane()" function
+///
+/// Display a message in the message pane.
+///
+/// @param message Message to display.
+/// @param [highlight] The highlight group name.
+/// @param [open] Opens the message page if it's not already in a window.
+///
+/// @return Nothing.
+static void f_msgpane(typval_T *argvars, typval_T *rettv)
+{
+  char_u *msg = get_tv_string(&argvars[0]);
+  int attr = 0;
+
+  if (argvars[1].v_type != VAR_UNKNOWN) {
+    char_u *attrName = get_tv_string(&argvars[1]);
+    int hl_id = syn_name2id(attrName);
+    attr = syn_id2attr(hl_id);
```

Comment:
ASAN finds a problem with `:echo msgpane("asdf", 0,0)`. Parameter values not checked?
```c
==4831==ERROR: AddressSanitizer: heap-buffer-overflow on address 0x62d00023a3b8 at pc 0x0000016d5ac9 bp 0x7ffc8214cf40 sp 0x7ffc8214cf38
READ of size 4 at 0x62d00023a3b8 thread T0
    #0 0x16d5ac8 in syn_id2attr /home/oni-link/git/neovim/src/nvim/syntax.c:7235:15
    #1 0x89857c in f_msgpane /home/oni-link/git/neovim/src/nvim/eval.c:13071:12
    #2 0x794e63 in call_func /home/oni-link/git/neovim/src/nvim/eval.c:7310:11
    #3 0x7a8da4 in get_func_tv /home/oni-link/git/neovim/src/nvim/eval.c:7159:11
    #4 0x839119 in eval7 /home/oni-link/git/neovim/src/nvim/eval.c:4277:15
    #5 0x834c45 in eval6 /home/oni-link/git/neovim/src/nvim/eval.c:3998:7
    #6 0x8318d9 in eval5 /home/oni-link/git/neovim/src/nvim/eval.c:3850:7
    #7 0x82ce51 in eval4 /home/oni-link/git/neovim/src/nvim/eval.c:3602:7
    #8 0x82c3e3 in eval3 /home/oni-link/git/neovim/src/nvim/eval.c:3524:7
    #9 0x82b973 in eval2 /home/oni-link/git/neovim/src/nvim/eval.c:3461:7
    #10 0x78d952 in eval1 /home/oni-link/git/neovim/src/nvim/eval.c:3394:7
    #11 0x7eb20e in ex_echo /home/oni-link/git/neovim/src/nvim/eval.c:19415:9
    #12 0xa990d0 in do_one_cmd /home/oni-link/git/neovim/src/nvim/ex_docmd.c:2193:5
    #13 0xa77c17 in do_cmdline /home/oni-link/git/neovim/src/nvim/ex_docmd.c:601:20
    #14 0x100c1fb in nv_colon /home/oni-link/git/neovim/src/nvim/normal.c:4489:18
    #15 0xff3743 in normal_execute /home/oni-link/git/neovim/src/nvim/normal.c:1144:3
    #16 0x1684d10 in state_enter /home/oni-link/git/neovim/src/nvim/state.c:56:26
    #17 0xfa7ab0 in normal_enter /home/oni-link/git/neovim/src/nvim/normal.c:464:3
    #18 0xd606c2 in main /home/oni-link/git/neovim/src/nvim/main.c:544:3
    #19 0x7f7cc90235af in __libc_start_main (/lib64/libc.so.6+0x205af)
    #20 0x445628 in _start (/home/oni-link/git/neovim/build/bin/nvim+0x445628)

0x62d00023a3b8 is located 72 bytes to the left of 34056-byte region [0x62d00023a400,0x62d000242908)
allocated by thread T0 here:
    #0 0x4f8fb8 in realloc /home/oni-link/git/llvm/projects/compiler-rt/lib/asan/asan_malloc_linux.cc:77
    #1 0xe8385c in xrealloc /home/oni-link/git/neovim/src/nvim/memory.c:141:15
    #2 0xc84e33 in ga_grow /home/oni-link/git/neovim/src/nvim/garray.c:96:14
    #3 0x16c55ef in highlight_changed /home/oni-link/git/neovim/src/nvim/syntax.c:7376:3
    #4 0x1321b77 in update_screen /home/oni-link/git/neovim/src/nvim/screen.c:342:5
    #5 0x1069995 in normal_redraw /home/oni-link/git/neovim/src/nvim/normal.c:1263:5
    #6 0x106702c in normal_check /home/oni-link/git/neovim/src/nvim/normal.c:1346:5
    #7 0x1684723 in state_enter /home/oni-link/git/neovim/src/nvim/state.c:20:35
    #8 0xfa7ab0 in normal_enter /home/oni-link/git/neovim/src/nvim/normal.c:464:3
    #9 0xd606c2 in main /home/oni-link/git/neovim/src/nvim/main.c:544:3
    #10 0x7f7cc90235af in __libc_start_main (/lib64/libc.so.6+0x205af)

SUMMARY: AddressSanitizer: heap-buffer-overflow /home/oni-link/git/neovim/src/nvim/syntax.c:7235:15 in syn_id2attr
```